core/services/keeper/upkeep_executer.go

```diff
@@ -309,3 +294,31 @@ func (ex *UpkeepExecuter) turnBlockHashBinary(registry Registry, head *evmtypes.
 	binaryString := fmt.Sprintf("%b", hashAtHeight.Big())
 	return binaryString, nil
 }
+
+func buildJobSpec(
+	jb job.Job,
+	upkeep UpkeepRegistration,
+	ormConfig RegistryGasChecker,
+	exConfig RegistryGasChecker,
```

Comment:
nit: I feel it's a bit of overhead to define `RegistryGasChecker` as a type. Maybe we can just pass performGasOverhead and checkGasOverhead as separate params?

Reply:
I wasn't sure if those configs had different values or not. It's a dangerous road to walk by having duplicate values across type properties. It creates the assumption that the values could be different.

The simple solution here is, of course, to pass a single value to this new function and hope the unit test catches any foul play in the future. I'll do that for now.